src/lib/OS/SyscallHandler.cc

```diff
@@ -1,15 +1,811 @@
+      int64_t flag = 0;
+
+      // Don't process the syscall if the virtual address translation comes back
+      // wih a DATA_ABORT or IGNORED fault
+      uint64_t faultCode = simeng::OS::masks::faults::getFaultCode(filenamePtr);
+      if (faultCode == simeng::OS::masks::faults::pagetable::DATA_ABORT ||
+          faultCode == simeng::OS::masks::faults::pagetable::IGNORED) {
+        return concludeSyscall({}, true);
+      } else {
+        char* filename = new char[PATH_MAX_LEN];
+        return readStringThen(
+            filename, filenamePtr, PATH_MAX_LEN, [=](auto length) {
+              // Invoke the kernel
+              int64_t retval = faccessat(dfd, filename, mode, flag);
+              ProcessStateChange stateChange = {
+                  ChangeType::REPLACEMENT, {info.ret}, {retval}};
+              delete[] filename;
+              concludeSyscall(stateChange);
+            });
+      }
+      break;
+    }
+    case 56: {  // openat
+      int64_t dirfd = info.registerArguments[0].get<int64_t>();
+      uint64_t pathnamePtr = vAddrTranslation_(
+          info.registerArguments[1].get<uint64_t>(), info.processId);
+      int64_t flags = info.registerArguments[2].get<int64_t>();
+      uint16_t mode = info.registerArguments[3].get<uint16_t>();
+
+      // Don't process the syscall if the virtual address translation comes back
+      // wih a DATA_ABORT or IGNORED fault
+      uint64_t faultCode = simeng::OS::masks::faults::getFaultCode(pathnamePtr);
+      if (faultCode == simeng::OS::masks::faults::pagetable::DATA_ABORT ||
+          faultCode == simeng::OS::masks::faults::pagetable::IGNORED) {
+        return concludeSyscall({}, true);
+      } else {
+        char* pathname = new char[PATH_MAX_LEN];
+        return readStringThen(
+            pathname, pathnamePtr, PATH_MAX_LEN, [=](auto length) {
+              // Invoke the kernel
+              uint64_t retval = openat(dirfd, pathname, flags, mode);
+              ProcessStateChange stateChange = {
+                  ChangeType::REPLACEMENT, {info.ret}, {retval}};
+              delete[] pathname;
+              concludeSyscall(stateChange);
+            });
+      }
+      break;
+    }
+    case 57: {  // close
+      int64_t fd = info.registerArguments[0].get<int64_t>();
+      stateChange = {ChangeType::REPLACEMENT, {info.ret}, {close(fd)}};
+      break;
+    }
+    case 61: {  // getdents64
+      int64_t fd = info.registerArguments[0].get<int64_t>();
+      uint64_t bufPtr = vAddrTranslation_(
+          info.registerArguments[1].get<uint64_t>(), info.processId);
+      uint64_t count = info.registerArguments[2].get<uint64_t>();
+
+      // Don't process the syscall if the virtual address translation comes back
+      // wih a DATA_ABORT fault
+      uint64_t faultCode = simeng::OS::masks::faults::getFaultCode(bufPtr);
+      if (faultCode == simeng::OS::masks::faults::pagetable::DATA_ABORT) {
```

Comment:
why can't fault code be pagetable::IGNORED?

Reply:
That's checked within the readBufferThen body